### PR TITLE
MGMT-14649: Add feature support for OCI platform

### PR DESCRIPTION
**Pull request by eliorerz**
Set OCI validations as described in [feature-support matrix](https://docs.google.com/spreadsheets/d/1IJugmChJTKMBjdXHojEgfsyE8WT-dGbUpETn8fhuI7g/edit#gid=0) 

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

/cc @adriengentil @gamli75

### CODE REVIEW

File: internal/featuresupport/common.go [hunk 1]
```diff
@@ -68,3 +68,16 @@ func getActivatedFeatures(log logrus.FieldLogger, cluster *common.Cluster, infra
 
 	return activatedFeatures
 }
+
+func isPlatformActive(cluster *common.Cluster, clusterUpdateParams *models.V2ClusterUpdateParams, expectedPlatform models.PlatformType) bool {
```
Comment (gamli75) [inline internal/featuresupport/common.go:72]: please add a comment about what is this method

File: swagger.yaml [hunk 2]
```diff
@@ -3991,6 +3991,8 @@ definitions:
       - 'CLUSTER_MANAGED_NETWORKING'
       - 'USER_MANAGED_NETWORKING'
       - 'MINIMAL_ISO'
+      - 'FULL_ISO'
```
Comment (gamli75) [inline swagger.yaml:3994]: Why you did not add a new struct for full ISO?

**Comment by eliorerz** [comment]
@adriengentil @gamli75 For some reason it seems that the previous commit wasn't the latest one. I pushed the code with the missing parts

File: internal/featuresupport/feature_support_test.go [hunk 3]
```diff
@@ -504,6 +505,17 @@ var _ = Describe("V2ListFeatureSupportLevels API", func() {
 				Expect((*isNutanixIncompatibleWithUmn).getId()).To(Equal(umnFeature.getId()))
 			})
 
+			It("Features with restrictions - OCI and CMN", func() {
```
Comment (adriengentil) [inline internal/featuresupport/feature_support_test.go:508]: Should we add a similar test for OCI + full iso ?

**Review by adriengentil — COMMENTED** [review]
Just one question, otherwise it looks good to me!

File: internal/featuresupport/feature_support_test.go (hunk 3 above)
Comment (eliorerz) [reply internal/featuresupport/feature_support_test.go:508]: I'll add one also for OCI + Full ISO

Comment (eliorerz) [reply internal/featuresupport/feature_support_test.go:507]: Done!

**Review by avishayt — APPROVED** [review]

**Review by gamli75 — APPROVED** [review]

**Pull request merged**